### ISSUE TITLE
The loss function choice in parameter objective

### ISSUE THREAD

**Issue by bastia1989** [dmlc/xgboost#3949]
In short, does `binary:logistic` and `reg:logistic` use different loss function, i.e., objective function? I think for binary classification, that the loss function should be the sum of cross entropy across all samples, right? The same question for `multi:softmax` and `multi:softprob`. Do they only differ in the output (one outputs classes and the other gives probability)?

Meanwhile, see the [answer by the author](https://github.com/dmlc/xgboost/issues/521#issuecomment-144453618), what's the point to use the `rmse`  metric for binary classification learner?

**Comment by KeruiZhu** [comment]
I think the author means that reg:logistic use rmse.

**Comment by bastia1989** [comment]
> I think the author means that reg:logistic use rmse.

Does that mean the cost or loss functions for `binary:logistic` and `reg:logistic` are different? If so, what's the point using `rmse` in logistic regression?

**Comment by thvasilo** [comment]
As far as I can tell from the [code](https://github.com/dmlc/xgboost/blob/84ab74f3a56739829b03161fb9c249f3a760a518/src/objective/regression_loss.h#L73) they are exactly the same. 

I assume the purpose is to have them as a convenient way to report RMSE in the evaluation stage of the learning, but this should not affect the actual learning.

In other words,  I think the end model from running these two objectives should be the same, the only difference would be the reported metrics in the evaluation while learning.

**Comment by bastia1989** [comment]
> As far as I can tell from the [code](https://github.com/dmlc/xgboost/blob/84ab74f3a56739829b03161fb9c249f3a760a518/src/objective/regression_loss.h#L73) they are exactly the same.
> 
> I assume the purpose is to have them as a convenient way to report RMSE in the evaluation stage of the learning, but this should not affect the actual learning.
> 
> In other words, I think the end model from running these two objectives should be the same, the only difference would be the reported metrics in the evaluation while learning.

Sounds quite reasonable. I need to try if forcing xgb.XGBClassifier to use `reg:logistic` and 'logloss' togeter will cause any difference.
